### PR TITLE
Update dependencies, modernize some bits of code

### LINKED ISSUE #43: Update scipy / opencv / matplotlib / numpy versions
In d8f4b88 we relaxed the scipy version requirement a bit, but `scipy==1.4.1` still only has wheels for python3.8 and below (see available wheels at https://pypi.org/project/scipy/1.4.1/#files). The latest version of scipy currently is `1.13.0`, which has wheels for python3.9 to python3.12. We should try just unpinning the scipy version to let pip install the latest version that's compatible with the version of python the user is trying to use. If unpinning breaks things, there might be some work to do to get everything to run with a more modern python version. Maybe aim for 3.10 or so.

This will also fix https://github.com/NeLy-EPFL/DeepFly3D/security/dependabot/2

I can get around to doing this sometime, though @azmaite if you're testing/using this code soon feel free to give the updated scipy a try.

### PR DESCRIPTION
**Pull request by jasper-tms**
Closes #43

### CODE REVIEW

**Review by tkclam — APPROVED** [review]

**Comment by jasper-tms** [comment]
Hi @Dominic-DallOsto I'd be ready to merge these changes. I've confirmed that the results videos that this branch produces (running updated versions of python and many packeges) are visually identical to the ones generated by the master branch (running old versions of things).

File: df3d/core.py [hunk 1]
```diff
@@ -412,7 +424,7 @@ def setup_camera_ordering(self, camera_ordering) -> np.ndarray:
 
     def expand_videos(self):
         """expands video camera_x.mp4 into set of images camera_x_img_y.jpg"""
-        for vid in glob.glob(os.path.join(self.input_folder, "camera_*.mp4")):
+        for vid in glob.glob(os.path.join(self.input_folder, "camera_?.mp4")):
```
Comment (Dominic-DallOsto) [inline df3d/core.py:427]: I guess this limits us to only using up to 10 cameras, but from what I could tell that we use 7 cameras is quite baked in so I think this makes sense

**Review by Dominic-DallOsto — APPROVED** [review]

**Comment by Dominic-DallOsto** [comment]
Looks good!

I'm trying to setup automated testing so just to check how you went about manually testing: you ran the code on the sample data using the current master and this branch, and compared if they looked the same?

**Comment by jasper-tms** [comment]
Yep. And I've been running this branch on "real" data and getting reasonable/expected results on it.

**Pull request merged**